### PR TITLE
Fix Subscriber primary data channel

### PR DESCRIPTION
**Pull request by davidliu**

### CODE REVIEW

File: livekit-android-sdk/src/main/java/io/livekit/android/room/RTCEngine.kt [hunk 1]
```diff
@@ -179,20 +179,17 @@ internal constructor(
 
         if (joinResponse.subscriberPrimary) {
             // in subscriber primary mode, server side opens sub data channels.
-            publisherObserver.dataChannelListener = onDataChannel@{ dataChannel: DataChannel? ->
-                if (dataChannel == null) {
-                    return@onDataChannel
-                }
+            subscriberObserver.dataChannelListener = onDataChannel@{ dataChannel: DataChannel ->
```
Comment (davidliu) [inline livekit-android-sdk/src/main/java/io/livekit/android/room/RTCEngine.kt:182]: Primary cause of the bug here, was looking for the datachannels on the wrong thing.

Comment (boks1971) [reply livekit-android-sdk/src/main/java/io/livekit/android/room/RTCEngine.kt:182]: Are we not using the publisher data channel at all? If the client wants to send data over data channel, is it using the subscriber peer connection data channel to do send that data?

Comment (davidliu) [reply livekit-android-sdk/src/main/java/io/livekit/android/room/RTCEngine.kt:182]: The publisher data channel is used for sending data packets, and the subscriber channel is for receiving them.

**Review by boks1971 — APPROVED** [review]
lgtm!

**Pull request merged**